src/hv_anndata/manifoldmap.py

```diff
@@ -164,21 +168,21 @@ def create_manifoldmap_plot(
     cmap = config.get("cmap")
     title = config.get("title", "")
     responsive = config.get("responsive", True)
+    streams = config.get("streams", [])
 
     # Determine if color data is categorical
     if categorical is None:
         categorical = _is_categorical(color_data)
 
-    # Add a NaN category to handle and display data points with no category
+    # Set colormap and plot options based on data type
     if categorical:
+        # Add a NaN category to handle and display data points with no category
         color_data = np.where(
             color_data != color_data,
             "NaN",
             color_data,
         )  # np.nan != np.nan is True
 
-    # Set colormap and plot options based on data type
-    if categorical:
```

Comment:
Had to combine this and the previous if statement because pre-commit said the function was too complex. I think it should not change behavior if I'm not mistaken.

Reply:
Looks right!

Reply:
Usually I follow this lint by recognizing that a function does multiple almost independent things in a sequence (and often has one comment per step).

I then use the comments to name new helper functions

```py
def complex_function(things):
    # find a bleh if possible
    bleh = None
    for thang in things:
        ...

    # create default bleh if not found
    if bleh is not None:
        return bleh
    ...
    return ...
```

becomes:

```py
def complex_function(things):
    bleh = _find_bleh(things)
    return _default_bleh(bleh)

def _find_bleh(things):
    for thang in things:
        ...
    return None

def _default_bleh(bleh):
    if bleh is not None:
        return bleh
    ...
```